warp/Network/Wai/Handler/Warp.hs

```diff
@@ -371,3 +391,10 @@ setHTTP2Disabled y = y { settingsHTTP2Enabled = False }
 -- Since 3.0.10
 pauseTimeout :: Request -> IO ()
 pauseTimeout = fromMaybe (return ()) . Vault.lookup pauseTimeoutKey . vault
+
+-- | Getting file information from the inside cache.
+--   If file information is not available, an IO error is thrown.
+--
+-- Since 3.1.10
+getFileInfo :: Request -> FilePath -> IO FileInfo
+getFileInfo = fromMaybe (\_ -> throwIO (userError "getFileInfo")) . Vault.lookup getFileInfoKey . vault
```

Comment:
If that's the case, I think the comments here are misleading. It says that it will check inside the cache, and if file information is not available, it will throw an exception. Perhaps it should read:

> Getting file information from the inside cache.
> If information on the given file is not yet in the cache, information will be looked up via a `stat` syscall.
> If this function is used an a `Request` generated by a WAI backend besides Warp, it will throw an exception.